internal/relationtuple/definitions.go

```diff
@@ -26,619 +19,85 @@ type (
 		RelationTupleManager() Manager
 	}
 	Manager interface {
-		GetRelationTuples(ctx context.Context, query *RelationQuery, options ...x.PaginationOptionSetter) ([]*InternalRelationTuple, string, error)
-		WriteRelationTuples(ctx context.Context, rs ...*InternalRelationTuple) error
-		DeleteRelationTuples(ctx context.Context, rs ...*InternalRelationTuple) error
+		GetRelationTuples(ctx context.Context, query *RelationQuery, options ...x.PaginationOptionSetter) ([]*RelationTuple, string, error)
+		WriteRelationTuples(ctx context.Context, rs ...*RelationTuple) error
+		DeleteRelationTuples(ctx context.Context, rs ...*RelationTuple) error
 		DeleteAllRelationTuples(ctx context.Context, query *RelationQuery) error
-		TransactRelationTuples(ctx context.Context, insert []*InternalRelationTuple, delete []*InternalRelationTuple) error
-	}
-
-	RelationCollection struct {
-		protoRelations    []*rts.RelationTuple
-		internalRelations []*InternalRelationTuple
+		TransactRelationTuples(ctx context.Context, insert []*RelationTuple, delete []*RelationTuple) error
 	}
 	SubjectID struct {
-		ID string `json:"id"`
+		ID uuid.UUID `json:"id"`
+	}
+	RelationQuery struct {
+		Namespace *int32     `json:"namespace"`
+		Object    *uuid.UUID `json:"object"`
+		Relation  *string    `json:"relation"`
+		Subject   Subject    `json:"subject_id,omitempty"`
+	}
+	TupleData interface {
+		GetSubject() *rts.Subject
+		GetObject() string
+		GetNamespace() string
+		GetRelation() string
+	}
+	Subject interface {
+		Equals(Subject) bool
+		Hash() uuid.UUID
```

Comment:
I find the name `Hash` somewhat surprising, given it just returns an ID. Maybe we can call this `ID` or `UUID`?